vividus-plugin-mobile-app/src/main/java/org/vividus/mobileapp/action/TouchActions.java

```diff
@@ -150,9 +159,16 @@ public void swipeUntil(SwipeDirection direction, Duration swipeDuration, Rectang
          * - payload is {"strategy": "accessibility id", "selector": "selector"}
          * - see https://appium.io/docs/en/commands/mobile-command/#android-uiautomator2-only
          * */
-
-        int screenshotHeightToCompare = genericWebDriverManager.getSize().getHeight() / HEIGHT_DIVIDER;
-        int heightOffset = screenshotHeightToCompare * 2;
+        SearchContext searchContext = uiContext.getSearchContext();
+        Supplier<BufferedImage> screenShooter = () -> screenshotTaker
+                .takeAshotScreenshot(searchContext, Optional.empty()).getImage();
+        if (searchContext instanceof WebDriver)
+        {
+            int screenshotHeightToCompare = genericWebDriverManager.getSize().getHeight() / HEIGHT_DIVIDER;
+            int heightOffset = screenshotHeightToCompare * 2;
+            screenShooter = Suppliers.compose((BufferedImage i) -> i.getSubimage(0, i.getHeight() - heightOffset,
+                    i.getWidth(), screenshotHeightToCompare), screenShooter);
```

Comment:
does it really worth? couldn't it be simple `if` at old lines 169-171?

Reply:
- I wanted to avoid calls for size per-iteration or when context is set.